libgnucash/engine/Transaction.h

```diff
@@ -767,6 +767,12 @@ time64 xaccTransGetVoidTime(const Transaction *tr);
 void xaccTransDump (const Transaction *trans, const char *tag);
 #endif
 
+/** The xaccTransRecord() method iterates through the splits and
+ *  and record the non-currency equivalent prices in the price database.
+ */
+void xaccTransRecordPrice (Transaction *trans);
```

Comment:
Wouldn't it be better named `xaccTransUpdatePrices` ?

Reply:
I stole this code from John's original piece in split-register.c:record_price.  These are most likely new entries to be added to the database. So record is slightly more accurate than update? Let me know if I should rename it.

Reply:
So instead of copying you should move the code and replace all of the calls to `record_price` in `split-register.c` with `xaccTransRecordPrice`.

Reply:
Sure. Will do. Thanks!